docs/porting.md

```diff
@@ -0,0 +1,56 @@
+# Porting guidelines
+
+This is a list of steps which should help you on your probably not so easy journey of porting some mod:
+
+### 1. Setting up repository and build system
+1. Checkout any information in mod REAMDE/Wiki/Docs to find out if there are any special tasks/configs that need to be applied to the build
+2. Fork original repository to preserve commit history
+3. Apply build migration as explained in [migration guidelines](migration.md) on your fork
+
+### 2. Refining the fork
+Try to get rid of dependencies on concrete jars (usually in the `lib` folder) if any present. This way it will be much easier to change (upgrade/downgrade) your project dependencies, when needed.
+
+Check if they are in maven repository (usually authors put such information in the project readme), if it isn't accessible but project is open source with permissive enough license (e.g. MIT) - you still can publish them yourselves:
+   1. Fork the repository
+   2. Drop `jitpack.yml` and `.github/workflows/gradle.yml` in project root. You can find this file in this repository root.
+   3. Make sure everything builds from console by running `./gradlew clean setupCIWorkspace`
+   4. If all is fine/after fixing the errors - make a tag on Github or using console, this should trigger Github build hook and generate a release
+   4. Lookup forked repository on `https://jitpack.io/`
+   5. Find your release and click "Get it", which should scroll you down to the example of how to add the dependency (make sure you have jitpack repository in mod you are porting)
+   6. Checkout build log beside button you clicked to make sure it succeed
+
+Now, having dependency string you can just plug it into dependencies
+
+If you can't find any source code of the lib, don't despair, there is still a chance to distribute it. If original author has not restricted the distribution permissions (hence the jar ended up in repository of the mod you are porting) or you managed to obtain their permission - you can just upload it as a jar to jitpack, see [jitpack single file publishing thread](https://gist.github.com/jitpack-io/f928a858aa5da08ad9d9662f982da983) for that.
+
+There also may be a case where mod depends on another mod - then you'll need to port it first. (Yay, dependency hell! :D)
```

Comment:
There may also be a case where mods depend on other mods - then you'll need to port all dependencies first.